docs/extensibility.md

```diff
@@ -120,38 +120,64 @@ wp.hooks.addFilter(
 
 _Note:_ This filter must always be run on every page load, and not in your browser's developer tools console. Otherwise, a [block validation](https://wordpress.org/gutenberg/handbook/block-api/block-edit-save/#validation) error will occur the next time the post is edited. This is due to the fact that block validation occurs by verifying that the saved output matches what is stored in the post's content during editor initialization. So, if this filter does not exist when the editor loads, the block will be marked as invalid.
 
-## Adding a sidebar
-
-**Warning:** This is an experimental API, and is subject to change or even removal.
-
-### Registering a sidebar
+## Extending the editor's UI (Slot and Fill)
 
-`wp.editPost.__experimentalRegisterSidebar( name: string, settings: { title: string, render: function } )`
+Extending the editor's UI is done with in `registerPlugin` API, and allows you to define all your plugin's UI elements in one place.
 
-This method takes a sidebar `name` and a `settings` object, containing a title and a render function. The name should contain a namespace prefix (Example: my-plugin/my-custom-sidebar).
+`wp.editPost.registerPlugin( { name: string, render: function } )`
```

Comment:
It might be a good idea to make it a header in a similar way we do it in `data` module:

https://github.com/WordPress/gutenberg/blob/master/data/README.md#select-storename-string--object